scipy/stats/_mannwhitneyu.py

```diff
@@ -0,0 +1,430 @@
+    chance exceeds 5%, so we do not consider the results statistically
+    significant.
+
+    Suppose that, before seeing the data, we had hypothesized that females
+    would tend to be diagnosed at a younger age than males.
+    In that case, it would be natural to provide the female ages as the
+    first input, and we would have performed a one-sided test using
+    ``alternative = 'less'``: females are diagnosed at an age that is
+    stochastically less than that of males.
+
+    >>> res = mannwhitneyu(females, males, alternative="less", method="exact")
+    >>> print(res)
+    MannwhitneyuResult(statistic=3.0, pvalue=0.05555555555555555)
+
+    Again, the probability of getting a sufficiently low value of the
+    test statistic by chance under the null hypothesis is greater than 5%,
+    so we do not reject the null hypothesis in favor of our alternative.
+
+    If it is reasonable to assume that the means of samples from the
+    populations are normally distributed, we could have used a t-test to
+    perform the analysis.
+
+    >>> from scipy.stats import ttest_ind
+    >>> res = ttest_ind(females, males, alternative="less")
+    >>> print(res)
+    Ttest_indResult(statistic=-2.239334696520584, pvalue=0.030068441095757924)
+
+    Under this assumption, the *p*-value would be low enough to reject the
+    null hypothesis in favor of the alternative.
+
+    '''
+
+    x, y, use_continuity, alternative, axis_int, method = (
+        _mwu_input_validation(x, y, use_continuity, alternative, axis, method))
+
+    x, y, xy = _broadcast_concatenate(x, y, axis)
+
+    n1, n2 = x.shape[-1], y.shape[-1]
+
+    if method == "auto":
+        method = _mwu_choose_method(n1, n2, xy, method)
+
+    # Follows [2]
+    ranks = stats.rankdata(xy, axis=-1)  # method 2, step 1
+    R1 = ranks[..., :n1].sum(axis=-1)    # method 2, step 2
+    U1 = R1 - n1*(n1+1)/2                # method 2, step 3
+    U2 = n1 * n2 - U1                    # as U1 + U2 = n1 * n2
+
+    if alternative == "greater":
+        U, f = U1, 1  # U is the statistic to use for p-value, f is a factor
+    elif alternative == "less":
+        U, f = U2, 1  # Due to symmetry, use SF of U2 rather than CDF of U1
+    else:
+        U, f = np.maximum(U1, U2), 2  # multiply SF by two for two-sided test
+
+    if method == "exact":
+        p = _mwu_state.sf(U.astype(int), n1, n2)
+    elif method == "asymptotic":
+        z = _get_mwu_z(U, n1, n2, ranks, continuity=use_continuity)
+        p = stats.norm.sf(z)
+    p *= f
+
+    # Ensure that test statistic is not greater than 1
+    # This could happen for exact test when U = m*n/2
+    # Written to avoid dealing with floats and Nd arrays separately
+    # If you prefer to branch, please suggest specific code
+    pgt1 = p > 1
+    p *= np.logical_not(pgt1)  # zero entries > 1, preserve the rest
+    p += pgt1                  # add 1 to entries > 1, preserve the rest
```

Comment:
```suggestion
    # Written to avoid dealing with floats and Nd arrays separately
    # If you prefer to branch, please suggest specific code
    p = np.clip(p, 0, 1)
```

Not sure what I was thinking before, but I hope this will work.